core/services/ocr2/plugins/ocr2keeper/evm21/mercury/mercury.go

```diff
@@ -0,0 +1,122 @@
+package mercury
+
+import (
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/models"
+)
+
+const (
+	FeedIDs                  = "feedIDs"     // valid for v0.3
+	FeedIdHex                = "feedIdHex"   // valid for v0.2
+	BlockNumber              = "blockNumber" // valid for v0.2
+	Timestamp                = "timestamp"   // valid for v0.3
+	totalFastPluginRetries   = 5
+	totalMediumPluginRetries = 10
+)
+
+var GenerateHMACFn = func(method string, path string, body []byte, clientId string, secret string, ts int64) string {
+	bodyHash := sha256.New()
+	bodyHash.Write(body)
+	hashString := fmt.Sprintf("%s %s %s %s %d",
+		method,
+		path,
+		hex.EncodeToString(bodyHash.Sum(nil)),
+		clientId,
+		ts)
+	signedMessage := hmac.New(sha256.New, []byte(secret))
+	signedMessage.Write([]byte(hashString))
+	userHmac := hex.EncodeToString(signedMessage.Sum(nil))
+	return userHmac
+}
+
+// CalculateRetryConfig returns plugin retry interval based on how many times plugin has retried this work
+var CalculateRetryConfigFn = func(prk string, mercuryConfig MercuryConfigProvider) time.Duration {
+	var ri time.Duration
+	var retries int
+	totalAttempts, ok := mercuryConfig.GetPluginRetry(prk)
+	if ok {
+		retries = totalAttempts.(int)
+		if retries < totalFastPluginRetries {
+			ri = 1 * time.Second
+		} else if retries < totalMediumPluginRetries {
+			ri = 5 * time.Second
+		}
+		// if the core node has retried totalMediumPluginRetries times, do not set retry interval and plugin will use
+		// the default interval
+	} else {
+		ri = 1 * time.Second
```

Comment:
Instead of this else block, you could change line 44 to be`ri := 1 * time.Second`? Could we also rename `ri` to `interval` or `retryInterval`?

Reply:
i renamed the variable. this else check should be here bc ok means contains or not...